### PR TITLE
Add parallel job envs

### PR DESCRIPTION
**Pull request by JuanitoFatas**
Add these two from [Parallel Jobs](https://buildkite.com/docs/tutorials/parallel-builds#parallel-jobs).

CleanShot 2021-03-16 at 13 52 20@2x

https://docs-review-more-buildk-qonyjd.herokuapp.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-parallel-job

### CODE REVIEW

File: pages/pipelines/environment_variables.md.erb [hunk 1]
```diff
@@ -303,6 +303,18 @@ The organization name on Buildkite as used in URLs. The value cannot be modified
 
 Example: `"acme-inc"`
 
+<h3 class="h3-caps">BUILDKITE_PARALLEL_JOB</h3>
+
+The number of parallel jobs created from a parallel build step. Starting from 0. For a build step with `parallelism: 5`, the value would be 0, 1, 2, 3, and 4 respectively.
```
Comment (plaindocs) [inline pages/pipelines/environment_variables.md.erb:308]: So this is the "index" for a specific job created from a parallel step? 

We might need something like

The index of each parallel job created from a parallel build step. Starts from 0 and has a maximum of (total parallelism -1 ).  For a build step with `parallelism: 5`, the value for each job would be 0, 1, 2, 3, and 4 respectively.

File: pages/pipelines/environment_variables.md.erb [hunk 2]
```diff
@@ -303,6 +303,18 @@ The organization name on Buildkite as used in URLs. The value cannot be modified
...
+
+Example: `"0"`
+
+<h3 class="h3-caps">BUILDKITE_PARALLEL_JOB_COUNT</h3>
+
+The total number of parallel jobs created from a parallel build step. For a build step with `parallelism: 5`, the value is 5.
```
Comment (plaindocs) [inline pages/pipelines/environment_variables.md.erb:314]: This is the max parallel jobs right? There could be a case where there are only 2 parallel jobs running but there theoretically could be 5? Or is it the actual count of parallel jobs created?

**Review by plaindocs — CHANGES_REQUESTED** [review]
Thanks @JuanitoFatas !

Couple of questions inline.

File: pages/pipelines/environment_variables.md.erb (hunk 2 above)
Comment (JuanitoFatas) [reply pages/pipelines/environment_variables.md.erb:314]: If `parallelism` says 5 it should be 5 for that step. Otherwise, something went wrong 😅

**Review by plaindocs — APPROVED** [review]

**Pull request merged**